tests/testthat/test-create_task_makefile.R

```diff
@@ -0,0 +1,110 @@
+  task_makefile <- create_task_makefile(
+    task_plan, makefile=NULL,
+    file_extensions=c('ind'), packages=c('scipiper','mda.streams'))
+  expect_is(task_makefile, "character")
+  remake_list <- yaml::yaml.load(task_makefile)
+  expect_equal(tail(names(remake_list$targets), 1), "all_tasks.ind_promise")
+  expect_equal(head(names(remake_list$targets), 1), "all_tasks")
+  expect_equal(tail(remake_list$targets, 1)[[1]]$command, 
+               "combine_to_ind(I('states/log/all_tasks.ind'), 'states/log/AZ.ind', 'states/log/CA.ind', 'states/log/CO.ind')")
+})
+
+test_that("can create task_makefile with named target", {
+  task_makefile <- create_task_makefile(
+    task_plan, makefile=NULL, target_name = 'my_states',
+    file_extensions=c('ind'), packages=c('scipiper','mda.streams'))
+  remake_list <- yaml::yaml.load(task_makefile)
+  expect_equal(tail(names(remake_list$targets), 1), "my_states_promise")
+  expect_equal(head(names(remake_list$targets), 1), "my_states_all")
+  expect_equal(tail(remake_list$targets, 1)[[1]]$command, 
+               "combine_to_ind( 'states/log/AZ.ind', 'states/log/CA.ind', 'states/log/CO.ind')")
+})
+
+test_that("can create task_makefile with named target and remake_file", {
+  task_makefile <- create_task_makefile(
+    task_plan, makefile=file.path(tempdir(), 'states.yml'), target_name = 'my_states',
+    file_extensions=c('ind'), packages=c('scipiper','mda.streams'))
+  remake_list <- yaml::yaml.load_file(task_makefile)
+  expect_equal(tail(names(remake_list$targets), 1), "my_states_promise")
+  expect_equal(head(names(remake_list$targets), 1), "states")
+  expect_equal(tail(remake_list$targets, 1)[[1]]$command, 
+               "combine_to_ind( 'states/log/AZ.ind', 'states/log/CA.ind', 'states/log/CO.ind')")
+})
+
+test_that("can create task_makefile with named target and remake_file w/o promises", {
+  task_makefile <- create_task_makefile(
+    task_plan, makefile=file.path(tempdir(), 'states.yml'), target_name = 'my_states',
+    file_extensions=c('ind'), packages=c('scipiper','mda.streams'), as_promises = FALSE)
+  remake_list <- yaml::yaml.load_file(task_makefile)
+  expect_equal(tail(names(remake_list$targets), 1), "my_states")
+})
+
+test_that("can't create task_makefile with unequal target names and combiners", {
+  expect_error(task_makefile <- create_task_makefile(
+    task_plan, makefile=file.path(tempdir(), 'states.yml'), target_name = 'my_states',
+    file_extensions=c('ind'), packages=c('scipiper','mda.streams'), as_promises = FALSE, 
+    finalize_fun = c('combine_these','combine_those')))
+})
+
+test_that("task_makefiles with multiple combiners", {
+  task_makefile <- create_task_makefile(
+    task_plan, makefile=file.path(tempdir(), 'states.yml'), target_name = c('these_states','those_states'),
+    file_extensions=c('ind'), packages=c('scipiper','mda.streams'), as_promises = FALSE, 
+    finalize_fun = c('combine_these','combine_those'))
```

Comment:
Running this command generates these messages:
```
run all tasks with
these_states:
  command: scmake(remake_file='C:\Users\aappling\AppData\Local\Temp\1\RtmpMb8XgC/states.yml')
run all tasks with
those_states:
  command: scmake(remake_file='C:\Users\aappling\AppData\Local\Temp\1\RtmpMb8XgC/states.yml')
```

On the one hand, maybe that's fine because (1) both suggestions are true and (2) does anybody read these messages anyway?

On the other hand, it'd be a little nicer if the message for how to "run all tasks" didn't recommend scmaking the same target twice, wouldn't it? Something like:
```
Run all tasks and finalizers with:
these_states:
  command: scmake(I('these_states_promise'), remake_file='C:/Users/aappling/Documents/states.yml')
those_states:
  command: scmake(I('those_states_promise'), remake_file='C:/Users/aappling/Documents/states.yml')
```

I _think_ I've implemented something workable at https://github.com/aappling-usgs/scipiper/commit/bd952c530d6d224290193269c35e1a7043a65122, but I haven't tested it for all the possible combinations of target_name, as_promises, and whatever else is relevant. We could push this to a separate issue if needed.

Reply:
I punted on this one, as handling all of those different combinations (e.g., targets, promises, etc) seemed time consuming and error prone, and I wasn't sure we wouldn't want to change the implementation after review or after playing with it.

Reply:
Ok - I tried to PR your branch to mine, but you have pitfalls vignette commits in there, so I stole your code for the message instead...lost attribution! 👎 but it looks good to me 💯